packages/@aws-cdk/aws-ivs/lib/stream-key.ts

```diff
@@ -0,0 +1,69 @@
+import * as core from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { IChannel } from './channel';
+import { CfnStreamKey } from './ivs.generated';
+
+/**
+ * Represents an IVS Stream Key
+ */
+export interface IStreamKey extends core.IResource {
+  /**
+   * The stream-key ARN. For example: arn:aws:ivs:us-west-2:123456789012:stream-key/g1H2I3j4k5L6
+   *
+   * @attribute
+   */
+  readonly streamKeyArn: string;
+}
+
+/**
+ * Reference to a new or existing IVS Stream Key
+ */
+abstract class StreamKeyBase extends core.Resource implements IStreamKey {
```

Comment:
Can skip the `Base` class if it's not providing any value. It can always be added later.

Reply:
Yeah I was just getting ahead of things really, is it better to go back and remove it? Or is it ok to leave it.

Reply:
I'd prefer to remove it; my default is less code -- that achieves the same purpose and is equally readable -- is better. I don't think the `StreamKeyBase` is adding any readability or functionality.